processor/configuration/src/main/java/com/redhat/service/smartevents/processor/sources/slack/SlackSourceValidator.java

```diff
@@ -0,0 +1,33 @@
+package com.redhat.service.smartevents.processor.sources.slack;
+
+import java.util.Map;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import com.redhat.service.smartevents.infra.models.actions.Source;
+import com.redhat.service.smartevents.infra.validations.ValidationResult;
+import com.redhat.service.smartevents.processor.sources.SourceValidator;
+
+@ApplicationScoped
+public class SlackSourceValidator implements SlackSource, SourceValidator {
+    public static final String INVALID_CHANNEL_MESSAGE =
+            "The supplied " + CHANNEL_PARAM + " parameter is not valid";
+
+    public static final String INVALID_TOKEN_MESSAGE =
+            "The supplied " + TOKEN_PARAM + " parameter is not valid";
+
+    @Override
+    public ValidationResult isValid(Source source) {
+        Map<String, String> parameters = source.getParameters();
+
+        if (!parameters.containsKey(CHANNEL_PARAM) || parameters.get(CHANNEL_PARAM).isEmpty()) {
+            return ValidationResult.invalid(INVALID_CHANNEL_MESSAGE);
+        }
+
+        if (!parameters.containsKey(TOKEN_PARAM) || parameters.get(TOKEN_PARAM).isEmpty()) {
```

Comment:
Storing this value in plaintext in the DB is fine for this first iteration to get things working. However, we should drop that as soon as possible. https://issues.redhat.com/browse/MGDOBR-59 captures the requirement to do that and is scheduled for next sprint.